ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/KeyConsistencyValidation.scala

```diff
@@ -68,47 +67,42 @@ private[keys] object KeyConsistencyValidation {
       key: Option[Node.KeyWithMaintainers[Value[ContractId]]],
       targetContractId: Option[ContractId],
       templateId: TypeConName,
-      keyValidationState: KeyValidationState,
-  ): KeyValidationStatus =
+      submittedContractKeysToContractIds: Map[DamlContractKey, Option[RawContractId]],
+  ): KeyConsistencyValidationStatus =
     key match {
-      case None => Right(keyValidationState)
+      case None => Right(submittedContractKeysToContractIds)
       case Some(submittedKeyWithMaintainers) =>
         val submittedDamlContractKey =
           Conversions.encodeContractKey(templateId, submittedKeyWithMaintainers.key)
-        val newKeyValidationState =
-          keyValidationState + ConsistencyKeyValidationState(
-            submittedContractKeysToContractIds = Map(
-              submittedDamlContractKey ->
-                targetContractId.map(_.coid)
+        val newSubmittedContractKeysToContractIds =
```

Comment:
@miklos-da this was just broken beforehand already. The `+` operator takes the second activeStateKeys set which means that here you always set it to an empty set. I added a testcase for this that fails on main (the two local key conflicts).

Rather than trying to fix this, I’ve just scraped the whole use of `+`. imho the new code is simpler and easier to get right but if you feel strongly about this, I can go back to the other solution. A fix would probably be for `+` to take the union instead. That doesn’t really make sense since a later state should always have a superset but given the way the code is setup here, that’s not the case.

Reply:
I don't feel strongly about this, I agree `+` didn't really help the purpose.